### ISSUE TITLE
get_wf_bulk_modulus fails if called without previous OptimizeFW

### ISSUE THREAD

**Issue by MichaelWolloch** [hackingmaterials/atomate#445]
> Please submit help issues to:
> https://hackingmaterials.discourse.group/c/atomate
> 
> The Github issues is no longer used except for internal development purposes.
> 
> If you are unable to use the Discourse forum, you may submit an issue here, but you must **clearly** state the reason you are unable to use the Discourse forum in your ticket. Otherwise, your issue will be **closed** without response.

I am sorry that I am writing about this here, but nobody reacted to my post on the Discourse forum which can be found [here](https://matsci.org/t/errors-in-get-wf-bulk-modulus/4138) (posted on July 4th).

To not copy everything from the forum post, I will be quick here:
get_wf_bulk_modulus fails if run _without_ running an OptimizeFW first. The workflow apparently expects a previous calculation. However, I want to use this to converge my energy-cutoff and do not want to pre-relax the system. I also do not see the reason why, if a OptimizeFW run has to be done before, it cannot be included in the workflow?

A quick fix is to set `copy_vasp_outputs=False` in the call to `get_wf_deformations` in line 58 of `atomate.vasp.workflows.base.bulk_modulus.py` and replace line 646 of `atomate.vasp.firetasks.parse_outputs.FitEOSToDb` (`d = mmdb.collection.find_one({"task_label": "{} structure optimization".format(tag)})`) with something like `d = mmdb.collection.find_one({"task_label": {"$regex": "{} bulk_modulus*".format(tag)}})`

Please advise if this is the intended behaviour. If not, please consider patching this by calling `get_wf_deformations` with the `copy_vasp_outputs=False` flag and changing  `atomate.vasp.firetasks.parse_outputs.FitEOSToDb`  accordingly (although my proposed solution seems not ideal in that case).

Thanks a lot, Michael

**Comment by utf** [comment]
Hi Michael,

Apologies for the delay responding to your issue. This is definitely a bug in atomate. Thank you for bringing it to our attention. I've addressed the issue in #465. 

The pull request thread gives more details on the fix but you should be able to run the exact code you posted in your message on the MatSci forum and have the workflow run successfully. 

If you could try using the latest atomate version from GitHub (rather than pip) and let me know whether this fixes your problem that would be much appreciated.

Best,
Alex

**Comment by MichaelWolloch** [comment]
Dear Alex,

I just discovered a problem that might have been also present in the old version. I never encountered it, because I pass a vasp_input_set, but if you do not, `atomate.vasp.firetasks.write_inputs.WriteTransmutedStructureIOSet` crashes with the following error:
~~~
2020-10-06 12:49:00,779 INFO Task started: {{atomate.vasp.firetasks.write_inputs.WriteTransmutedStructureIOSet}}.
Traceback (most recent call last):
  File "/home/michael/miniconda3/envs/Atomate_test/lib/python3.7/site-packages/fireworks/core/rocket.py", line 262, in run
    m_action = t.run_task(my_spec)
  File "/home/michael/miniconda3/envs/Atomate_test/atomate/atomate/vasp/firetasks/write_inputs.py", line 664, in run_task
    vis.write_input(".", potcar_spec=potcar_spec)
  File "/home/michael/miniconda3/envs/Atomate_test/lib/python3.7/site-packages/pymatgen/io/vasp/sets.py", line 693, in write_input
    zip_output=zip_output
  File "/home/michael/miniconda3/envs/Atomate_test/lib/python3.7/site-packages/pymatgen/io/vasp/sets.py", line 205, in write_input
    vinput = self.get_vasp_input()
  File "/home/michael/miniconda3/envs/Atomate_test/lib/python3.7/site-packages/pymatgen/io/vasp/sets.py", line 164, in get_vasp_input
    incar=self.incar,
  File "/home/michael/miniconda3/envs/Atomate_test/lib/python3.7/site-packages/pymatgen/io/vasp/sets.py", line 974, in incar
    self.kwargs.get("user_incar_settings", {}).keys()
AttributeError: 'NoneType' object has no attribute 'keys'
~~~

Since the default user_incar_settings get passed from `get_wf_bulk_modulus` are `None`, the self.kwargs.get("user_incar_settings", {}) method in line 794 of `pymatgen.io.vasp.sets` returns `None`,when it should return an empty dictionary to run the `.key()` method on.

In summary:
**I think the default for `user_incar_settings` in `get_wf_deformations` should thus be not `None`, but `{}`. Then the workflow runs fine with the defaults!**
(Maybe the defaults for vasp_cmd and db_file should also be changed to use env_chk?)

**Comment by utf** [comment]
Thanks for raising this issue.

This is actually a bug in pymatgen, as I believe that passing `user_incar_settings=None` to `MPStaticSet` should be fine. Indeed, `user_incar_settings=None` is actually the default value for the `DictSet` base class. Why this is happening is something to do with the `MPStaticSet` constructor and not completely obvious to me at this point.

I will update this thread when the fix is in pymatgen.
